tests/test_exonum_client.py

```diff
@@ -62,14 +87,68 @@ def mock_requests_get(url, params=None):
         # Proto sources for the supervisor service:
         (proto_sources_endpoint, "{'artifact': 'exonum-supervisor:0.11.0'}"): proto_sources_response("supervisor"),
         # System endpoints:
-        (healthcheck_endpoint, "None"): ok_response(),
-        (stats_endpoint, "None"): ok_response(),
-        (user_agent_endpoint, "None"): ok_response(),
+        # public
+        (healthcheck_endpoint, "None"): mock_response(200),
+        (stats_endpoint, "None"): mock_response(200),
+        (user_agent_endpoint, "None"): mock_response(200),
+        # private
+        (peers_endpoint, "None"): mock_response(200),
+        (consensus_endpoint, "None"): mock_response(200),
+        (network_endpoint, "None"): mock_response(200),
     }
 
+    # Explorer endpoints
+    if url == block_endpoint:
+        content = None
+        status_code = 200
+
+        if not isinstance(params["height"], int) or params["height"] < 0:
+            status_code = 400
+        else:
+            content = {"height": params["height"]}
+
+        responses.update({(block_endpoint, str(params)): mock_response(status_code, content)})
+    if url == blocks_endpoint:
+        content = None
+        status_code = 200
+
+        if not isinstance(params["count"], int) or params["count"] < 0:
+            status_code = 400
+        elif "earliest" in params and "latest" in params and params["latest"] - params["earliest"] < 0:
+            status_code = 200
+
+        responses.update({(blocks_endpoint, str(params)): mock_response(status_code, content)})
+    if url == transactions_endpoint:
+        content = None
+        status_code = 200
+
+        if not isinstance(params["hash"], str) or not params["hash"].isalnum():
+            status_code = 400
+
+        responses.update({(transactions_endpoint, str(params)): mock_response(status_code, content)})
+
     return responses[(url, str(params))]
 
 
+def mock_requests_post(url, data=None, headers=None):
+    exonum_public_base = EXONUM_URL_BASE.format(EXONUM_PROTO, EXONUM_IP, EXONUM_PUBLIC_PORT)
+    _exonum_private_base = EXONUM_URL_BASE.format(EXONUM_PROTO, EXONUM_IP, EXONUM_PRIVATE_PORT)
+
+    transactions_endpoint = exonum_public_base + EXPLORER_ENDPOINT_POSTFIX.format("transactions")
+    shutdown_endpoint = _exonum_private_base + SYSTEM_ENDPOINT_POSTFIX.format("shutdown")
+    consensus_endpoint = _exonum_private_base + SYSTEM_ENDPOINT_POSTFIX.format("consensus_enabled")
+    peers_endpoint = _exonum_private_base + SYSTEM_ENDPOINT_POSTFIX.format("peers")
+
+    if url == transactions_endpoint:
```

Comment:
Nit: this can be simplified as `if url in [transactions_endpoint, ..]:`.

Reply:
I was thinking about that. But what if someone will decite to extend the coverage beyond the 200 response for some endpoints?

Reply:
When this will become the case, we'll think about it ;)

Reply:
OK, I'll simplify the mentioned piece of code )